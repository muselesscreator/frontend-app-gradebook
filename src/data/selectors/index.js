/* eslint-disable import/no-named-as-default-member, import/no-self-import */
import { StrictDict } from 'utils';
import LmsApiService from 'data/services/LmsApiService';

import * as module from '.';
import app from './app';
import assignmentTypes from './assignmentTypes';
import cohorts from './cohorts';
import filters from './filters';
import grades from './grades';
import roles from './roles';
import special from './special';
import tracks from './tracks';

/**
 * lmsApiServiceArgs(state)
 * Returns common lms api service request args.
 * @param {object} state - redux state
 * @return {object} lms api query params object
 */
export const lmsApiServiceArgs = (state) => ({
  cohort: cohorts.getCohortNameById(state, filters.cohort(state)),
  assignment: filters.selectedAssignmentId(state),
  assignmentType: filters.assignmentType(state),
  assignmentGradeMin: grades.formatMinAssignmentGrade(
    filters.assignmentGradeMin(state),
    { assignmentId: filters.selectedAssignmentId(state) },
  ),
  assignmentGradeMax: grades.formatMaxAssignmentGrade(
    filters.assignmentGradeMax(state),
    { assignmentId: filters.selectedAssignmentId(state) },
  ),
  courseGradeMin: grades.formatMinCourseGrade(filters.courseGradeMin(state)),
  courseGradeMax: grades.formatMaxCourseGrade(filters.courseGradeMax(state)),
});

/**
 * gradeExportUrl(state, options)
 * Returns the output of getGradeExportCsvUrl, applying the current includeCourseRoleMembers
 * filter.
 * @param {object} state - redux state
 * @param {object} options - options object of the form ({ courseId })
 * @return {string} - generated grade export url
 */
export const gradeExportUrl = (state, { courseId }) => (
  LmsApiService.getGradeExportCsvUrl(courseId, {
    ...module.lmsApiServiceArgs(state),
    excludeCourseRoles: filters.includeCourseRoleMembers(state) ? '' : 'all',
  })
);

/**
 * interventionExportUrl(state, options)
 * Returns the output of getInterventionExportUrl.
 * @param {object} state - redux state
 * @param {object} options - options object of the form ({ courseId })
 * @return {string} - generated intervention export url
 */
export const interventionExportUrl = (state, { courseId }) => (
  LmsApiService.getInterventionExportCsvUrl(
    courseId,
    module.lmsApiServiceArgs(state),
  )
);

/**
 * getHeadings(state)
 * Returns the table headings given the current assignmentType and assignmentLabel filters.
 * @param {object} state - redux state
 * @return {string[]} - array of table headings
 */
export const getHeadings = (state) => grades.headingMapper(
  filters.assignmentType(state) || 'All',
  filters.selectedAssignmentLabel(state) || 'All',
)(grades.getExampleSectionBreakdown(state));

/**
 * showBulkManagement(state, options)
 * Returns true iff the user has special access or bulk management is configured to be available
 * and the course has a masters track.
 * @param {object} state - redux state
 * @param {object} options - options object of the form ({ courseId })
 * @return {bool} - should show bulk management controls?
 */
export const showBulkManagement = (state, { courseId }) => (
  special.hasSpecialBulkManagementAccess(courseId)
  || (tracks.stateHasMastersTrack(state) && state.config.bulkManagementAvailable)
);

/**
 * shouldShowSpinner(state)
 * Returns true iff the user can view the gradebook and grades.showSpinner is true.
 * @param {object} state - redux state
 * @return {bool} - should show spinner?
 */
export const shouldShowSpinner = (state) => (
  roles.canUserViewGradebook(state)
  && grades.showSpinner(state)
);

/**
 * showBulkManagement(state, options)
 * Returns true iff the user has special access or bulk management is configured to be available
 * and the course has a masters track.
 * @param {object} state - redux state
 * @return {bool} - should show bulk management controls?
 */
export const showBulkManagement = (state) => (
  special.hasSpecialBulkManagementAccess(app.courseId(state))
  || (tracks.stateHasMastersTrack(state) && state.config.bulkManagementAvailable)
);

export default StrictDict({
  root: StrictDict({
    getHeadings,
    gradeExportUrl,
    interventionExportUrl,
    shouldShowSpinner,
  }),
  app,
  assignmentTypes,
  cohorts,
  filters,
  grades,
  roles,
  special,
  tracks,
});
