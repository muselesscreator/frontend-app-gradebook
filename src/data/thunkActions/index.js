import { StrictDict } from 'utils';
import app from './app';
import assignmentTypes from './assignmentTypes';
import cohorts from './cohorts';
import filters from './filters';
import grades from './grades';
import roles from './roles';
import tracks from './tracks';

export default StrictDict({
  app,
  assignmentTypes,
  cohorts,
  filters,
  grades,
  roles,
  tracks,
});
