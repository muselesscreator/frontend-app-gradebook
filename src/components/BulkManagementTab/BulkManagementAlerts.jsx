/* eslint-disable react/sort-comp, react/button-has-type */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { Alert } from '@edx/paragon';

import * as appConstants from 'data/constants/app';
import selectors from 'data/selectors';

const { messages: { BulkManagementTab: messages } } = appConstants;

export const BulkManagementAlerts = ({ bulkImportError, uploadSuccess }) => (
  <>
    <Alert
      variant="danger"
      show={!!bulkImportError}
      dismissible={false}
    >
      {bulkImportError}
    </Alert>
    <Alert
      variant="success"
      show={uploadSuccess}
      dismissible={false}
    >
      {messages.successDialog}
    </Alert>
  </>
);

BulkManagementAlerts.defaultProps = {
  bulkImportError: '',
  uploadSuccess: false,
};

BulkManagementAlerts.propTypes = {
  // redux
  bulkImportError: PropTypes.string,
  uploadSuccess: PropTypes.bool,
};

export const mapStateToProps = (state) => ({
  bulkImportError: selectors.grades.bulkImportError(state),
  uploadSuccess: selectors.grades.uploadSuccess(state),
});

export default connect(mapStateToProps)(BulkManagementAlerts);
