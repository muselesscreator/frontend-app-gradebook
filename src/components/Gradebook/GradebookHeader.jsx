import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { configuration } from 'config';
import selectors from 'data/selectors';

export class GradebookHeader extends React.Component {
  lmsInstructorDashboardUrl = courseId => (
    `${configuration.LMS_BASE_URL}/courses/${courseId}/instructor`
  );

  render() {
    return (
      <div className="gradebook-header">
        <a
          href={this.lmsInstructorDashboardUrl(this.props.courseId)}
          className="mb-3"
        >
          <span aria-hidden="true">{'<< '}</span> Back to Dashboard
        </a>
        <h1>Gradebook</h1>
        <h3> {this.props.courseId}</h3>
        {this.props.areGradesFrozen
          && (
          <div className="alert alert-warning" role="alert">
            The grades for this course are now frozen. Editing of grades is no longer allowed.
          </div>
          )}
        {(this.props.canUserViewGradebook === false) && (
          <div className="alert alert-warning" role="alert">
            You are not authorized to view the gradebook for this course.
          </div>
        )}
      </div>
    );
  }
}

GradebookHeader.defaultProps = {
  courseId: '',
  // redux
  areGradesFrozen: false,
  canUserViewGradebook: false,
};

GradebookHeader.propTypes = {
  courseId: PropTypes.string,
  // redux
  areGradesFrozen: PropTypes.bool,
  canUserViewGradebook: PropTypes.bool,
};

export const mapStateToProps = (state) => ({
  areGradesFrozen: selectors.assignmentTypes.areGradesFrozen(state),
  canUserViewGradebook: selectors.roles.canUserViewGradebook(state),
});

export default connect(mapStateToProps)(GradebookHeader);
