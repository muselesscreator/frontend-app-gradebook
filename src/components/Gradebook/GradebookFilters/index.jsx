/* eslint-disable react/sort-comp, import/no-named-as-default */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { Collapsible, Form } from '@edx/paragon';

import actions from 'data/actions';
import selectors from 'data/selectors';
import thunkActions from 'data/thunkActions';

import AssignmentTypeFilter from './AssignmentTypeFilter';
import AssignmentFilter from './AssignmentFilter';
import AssignmentGradeFilter from './AssignmentGradeFilter';
import CourseGradeFilter from './CourseGradeFilter';
import StudentGroupsFilter from './StudentGroupsFilter';

export class GradebookFilters extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      includeCourseRoleMembers: this.props.includeCourseRoleMembers,
    };
    this.handleIncludeTeamMembersChange = this.handleIncludeTeamMembersChange.bind(this);
  }

  handleIncludeTeamMembersChange(event) {
    const includeCourseRoleMembers = event.target.checked;
    this.setState({ includeCourseRoleMembers });
    this.props.updateIncludeCourseRoleMembers(includeCourseRoleMembers);
    this.props.fetchGrades();
    this.props.updateQueryParams({ includeCourseRoleMembers });
  }

  collapsibleGroup = (title, content) => (
    <Collapsible title={title} defaultOpen className="filter-group mb-3">
      {content}
    </Collapsible>
  );

  render() {
    const {
      updateQueryParams,
    } = this.props;
    return (
      <>
        {this.collapsibleGroup('Assignments', (
          <div>
            <AssignmentTypeFilter updateQueryParams={updateQueryParams} />
            <AssignmentFilter updateQueryParams={updateQueryParams} />
            <AssignmentGradeFilter updateQueryParams={updateQueryParams} />
          </div>
        ))}
        {this.collapsibleGroup('Overall Grade', (
          <CourseGradeFilter updateQueryParams={updateQueryParams} />
        ))}
        {this.collapsibleGroup('Student Groups', (
          <StudentGroupsFilter updateQueryParams={updateQueryParams} />
        ))}
        {this.collapsibleGroup('Include Course Team Members', (
          <Form.Checkbox
            checked={this.state.includeCourseRoleMembers}
            onChange={this.handleIncludeTeamMembersChange}
          >
            Include Course Team Members
          </Form.Checkbox>
        ))}
      </>
    );
  }
}
GradebookFilters.defaultProps = {
  includeCourseRoleMembers: false,
};
GradebookFilters.propTypes = {
  fetchGrades: PropTypes.func.isRequired,
  includeCourseRoleMembers: PropTypes.bool,
  updateIncludeCourseRoleMembers: PropTypes.func.isRequired,
  updateQueryParams: PropTypes.func.isRequired,
};

export const mapStateToProps = (state) => ({
  includeCourseRoleMembers: selectors.filters.includeCourseRoleMembers(state),
});

export const mapDispatchToProps = {
  fetchGrades: thunkActions.grades.fetchGrades,
  updateIncludeCourseRoleMembers: actions.filters.update.includeCourseRoleMembers,
};

export default connect(mapStateToProps, mapDispatchToProps)(GradebookFilters);
