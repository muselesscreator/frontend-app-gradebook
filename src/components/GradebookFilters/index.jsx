/* eslint-disable react/sort-comp, import/no-named-as-default */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import {
  Collapsible,
  Icon,
  IconButton,
  Form,
} from '@edx/paragon';
import { Close } from '@edx/paragon/icons';

import actions from 'data/actions';
import selectors from 'data/selectors';
import thunkActions from 'data/thunkActions';

import AssignmentTypeFilter from './AssignmentTypeFilter';
import AssignmentFilter from './AssignmentFilter';
import AssignmentGradeFilter from './AssignmentGradeFilter';
import CourseGradeFilter from './CourseGradeFilter';
import StudentGroupsFilter from './StudentGroupsFilter';

export class GradebookFilters extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      includeCourseRoleMembers: this.props.includeCourseRoleMembers,
    };
    this.handleIncludeTeamMembersChange = this.handleIncludeTeamMembersChange.bind(this);
  }

  handleIncludeTeamMembersChange(event) {
    const includeCourseRoleMembers = event.target.checked;
    this.setState({ includeCourseRoleMembers });
    this.props.updateIncludeCourseRoleMembers(includeCourseRoleMembers);
    this.props.fetchGrades();
    this.props.updateQueryParams({ includeCourseRoleMembers });
  }

  collapsibleGroup = (title, content) => (
    <Collapsible title={title} defaultOpen className="filter-group mb-3">
      {content}
    </Collapsible>
  );

  render() {
    const {
      updateQueryParams,
    } = this.props;
    return (
      <>
        <div className="filter-sidebar-header">
          <h2><Icon className="fa fa-filter" /></h2>
          <IconButton
            className="p-1"
            onClick={this.props.closeMenu}
            iconAs={Icon}
            src={Close}
            alt="Close Filters"
            aria-label="Close Filters"
          />
        </div>

        {this.collapsibleGroup('Assignments', (
          <div>
            <AssignmentTypeFilter updateQueryParams={updateQueryParams} />
            <AssignmentFilter updateQueryParams={updateQueryParams} />
            <AssignmentGradeFilter updateQueryParams={updateQueryParams} />
          </div>
        ))}

        {this.collapsibleGroup('Overall Grade', (
          <CourseGradeFilter updateQueryParams={updateQueryParams} />
        ))}

        {this.collapsibleGroup('Student Groups', (
          <StudentGroupsFilter updateQueryParams={updateQueryParams} />
        ))}

        {this.collapsibleGroup('Include Course Team Members', (
          <Form.Checkbox
            checked={this.state.includeCourseRoleMembers}
            onChange={this.handleIncludeTeamMembersChange}
          >
            Include Course Team Members
          </Form.Checkbox>
        ))}
      </>
    );
  }
}
GradebookFilters.defaultProps = {
  includeCourseRoleMembers: false,
};
GradebookFilters.propTypes = {
  updateQueryParams: PropTypes.func.isRequired,
  // redux
  closeMenu: PropTypes.func.isRequired,
  fetchGrades: PropTypes.func.isRequired,
  includeCourseRoleMembers: PropTypes.bool,
  updateIncludeCourseRoleMembers: PropTypes.func.isRequired,
};

export const mapStateToProps = (state) => ({
  includeCourseRoleMembers: selectors.filters.includeCourseRoleMembers(state),
});

export const mapDispatchToProps = {
  closeMenu: thunkActions.app.filterMenu.close,
  fetchGrades: thunkActions.grades.fetchGrades,
  updateIncludeCourseRoleMembers: actions.filters.update.includeCourseRoleMembers,
};

export default connect(mapStateToProps, mapDispatchToProps)(GradebookFilters);
