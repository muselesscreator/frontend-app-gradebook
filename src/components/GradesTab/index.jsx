/* eslint-disable react/sort-comp, react/button-has-type, import/no-named-as-default */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import actions from 'data/actions';
import thunkActions from 'data/thunkActions';

import PageButtons from '../PageButtons';
import FilterBadges from '../FilterBadges';

import BulkManagementControls from './BulkManagementControls';
import EditModal from './EditModal';
import GradebookTable from './GradebookTable';
import SearchControls from './SearchControls';
import StatusAlerts from './StatusAlerts';
import SpinnerIcon from './SpinnerIcon';
import ScoreViewInput from './ScoreViewInput';
import UsersLabel from './UsersLabel';

export class GradesTab extends React.Component {
  constructor(props) {
    super(props);
    this.handleFilterBadgeClose = this.handleFilterBadgeClose.bind(this);
  }

  handleFilterBadgeClose(filterNames) {
    return () => {
      this.props.resetFilters(filterNames);
      this.props.updateQueryParams(filterNames.reduce(
        (obj, filterName) => ({ ...obj, [filterName]: false }),
        {},
      ));
      this.props.fetchGrades();
    };
  }

  render() {
    return (
      <>
        <SpinnerIcon />
        <SearchControls />
        <FilterBadges handleFilterBadgeClose={this.handleFilterBadgeClose} />
        <StatusAlerts />

        <h4>Step 2: View or Modify Individual Grades</h4>
        <UsersLabel />

        <div className="d-flex justify-content-between align-items-center mb-2">
          <ScoreViewInput />
          <BulkManagementControls />
        </div>

        <GradebookTable />

        <PageButtons />
        <p>* available for learners in the Master&apos;s track only</p>
        <EditModal />
      </>
    );
  }
}

GradesTab.defaultProps = {};

GradesTab.propTypes = {
  updateQueryParams: PropTypes.func.isRequired,

  // redux
  fetchGrades: PropTypes.func.isRequired,
  resetFilters: PropTypes.func.isRequired,
};

export const mapStateToProps = () => ({});

export const mapDispatchToProps = {
  fetchGrades: thunkActions.grades.fetchGrades,
  resetFilters: actions.filters.reset,
};

export default connect(mapStateToProps, mapDispatchToProps)(GradesTab);
