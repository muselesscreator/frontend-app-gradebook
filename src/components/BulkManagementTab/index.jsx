/* eslint-disable react/button-has-type, import/no-named-as-default */
import React from 'react';

import { messages } from 'data/constants/app';
import BulkManagementAlerts from './BulkManagementAlerts';
import FileUploadForm from './FileUploadForm';
import HistoryTable from './HistoryTable';

/**
 * <BulkManagementTab />
 * top-level view for managing uploads of bulk management override csvs.
 */
export const BulkManagementTab = () => (
  <div>
    <h4>{messages.BulkManagementTab.heading}</h4>
    <BulkManagementAlerts />
    <FileUploadForm />
    <HistoryTable />
  </div>
);

export default BulkManagementTab;
