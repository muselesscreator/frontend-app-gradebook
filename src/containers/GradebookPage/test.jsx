/* eslint-disable import/no-named-as-default */
import React from 'react';
import { shallow } from 'enzyme';
import queryString from 'query-string';

import selectors from 'data/selectors';
import thunkActions from 'data/thunkActions';

import { Tab, Tabs } from '@edx/paragon';

import GradebookFilters from 'components/GradebookFilters';
import GradebookHeader from 'components/GradebookHeader';
import GradesTab from 'components/GradesTab';
import BulkManagementTab from 'components/BulkManagementTab';

import { GradebookPage, mapStateToProps, mapDispatchToProps } from '.';

jest.mock('query-string', () => ({
  parse: jest.fn(val => ({ parsed: val })),
  stringify: (val) => `stringify: ${JSON.stringify(val, Object.keys(val).sort())}`,
}));

jest.mock('@edx/paragon', () => ({
  Tab: () => 'Tab',
  Tabs: () => 'Tabs',
}));
jest.mock('data/selectors', () => ({
  __esModule: true,
  default: {
    root: {
      showBulkManagement: (state) => ({ showBulkManagement: state }),
    },
  },
}));
jest.mock('data/thunkActions', () => ({
  __esModule: true,
  default: {
    app: { initialize: jest.fn() },
  },
}));

jest.mock('components/WithSidebar', () => 'WithSidebar');
jest.mock('components/GradebookHeader', () => 'GradebookHeader');
jest.mock('components/GradesTab', () => 'GradesTab');
jest.mock('components/GradebookFilters', () => 'GradebookFilters');
jest.mock('components/BulkManagementTab', () => 'BulkManagementTab');

describe('GradebookPage', () => {
  describe('component', () => {
    const courseId = 'a course';
    let el;
    const props = {
      location: {
        search: 'searchString',
      },
      match: { params: { courseId } },
    };
    beforeEach(() => {
      props.initializeApp = jest.fn();
      props.history = { push: jest.fn() };
    });
    test('snapshot - shows BulkManagementTab if showBulkManagement', () => {
      el = shallow(<GradebookPage {...props} showBulkManagement />);
      el.instance().updateQueryParams = jest.fn().mockName('updateQueryParams');
      expect(el.instance().render()).toMatchSnapshot();
    });
    test('snapshot - shows only GradesTab if showBulkManagement=false', () => {
      el = shallow(<GradebookPage {...props} />);
      el.instance().updateQueryParams = jest.fn().mockName('updateQueryParams');
      expect(el.instance().render()).toMatchSnapshot();
    });
    describe('render', () => {
      beforeEach(() => {
        el = shallow(<GradebookPage {...props} />);
      });
      describe('top-level WithSidebar', () => {
        test('sidebar from GradebookFilters, with updateQueryParams', () => {
          const { sidebar } = el.props();
          expect(sidebar).toEqual(
            <GradebookFilters updateQueryParams={el.instance().updateQueryParams} />,
          );
        });
      });
      describe('gradebook-content', () => {
        let content;
        let children;
        beforeEach(() => {
          content = el.props().children;
          children = content.props.children;
        });
        it('is wrapped in a div w/ px-3 gradebook-content classNames', () => {
          expect(content.type).toEqual('div');
          expect(content.props.className).toEqual('px-3 gradebook-content');
        });
        it('displays Gradebook header and then tabs', () => {
          expect(children[0]).toEqual(<GradebookHeader />);
        });
        it('displays tabs with only GradesTab if not showBulkManagement', () => {
          expect(shallow(children[1])).toEqual(shallow(
            <Tabs defaultActiveKey="grades">
              <Tab eventKey="grades" title="Grades">
                <GradesTab updateQueryParams={el.instance().updateQueryParams} />
              </Tab>
            </Tabs>,
          ));
        });
        it('displays tabs with grades and BulkManagement if showBulkManagement', () => {
          el = shallow(<GradebookPage {...props} showBulkManagement />);
          const tabs = el.props().children.props.children[1];
          expect(tabs).toEqual(
            <Tabs defaultActiveKey="grades">
              <Tab eventKey="grades" title="Grades">
                <GradesTab updateQueryParams={el.instance().updateQueryParams} />
              </Tab>
              <Tab eventKey="bulk_management" title="Bulk Management">
                <BulkManagementTab />
              </Tab>
            </Tabs>,
          );
        });
      });
    });
    describe('behavior', () => {
      beforeEach(() => {
        el = shallow(<GradebookPage {...props} />, { disableLifecucleMethods: true });
      });
      describe('componentDidMount', () => {
        test('initializes app with courseId and urlQuery', () => {
          el.instance().componentDidMount();
          expect(props.initializeApp).toHaveBeenCalledWith(
            courseId,
            queryString.parse(props.location.search),
          );
        });
      });
      describe('updateQueryParams', () => {
        it('replaces values for truthy values', () => {
          queryString.parse.mockImplementation(key => ({ [key]: key }));
          const newKey = 'testKey';
          const val1 = 'VALUE';
          const val2 = 'VALTWO!!';
          const args = { [newKey]: val1, [props.location.search]: val2 };
          el.instance().updateQueryParams(args);
          expect(props.history.push).toHaveBeenCalledWith(`?${queryString.stringify(args)}`);
        });
        it('clears values for non-truthy values', () => {
          queryString.parse.mockImplementation(key => ({ [key]: key }));
          const newKey = 'testKey';
          const val1 = 'VALUE';
          const val2 = false;
          const args = { [newKey]: val1, [props.location.search]: val2 };
          el.instance().updateQueryParams(args);
          expect(props.history.push).toHaveBeenCalledWith(
            `?${queryString.stringify({ [newKey]: val1 })}`,
          );
        });
      });
    });
  });
  describe('mapStateToProps', () => {
    let mapped;
    const testState = { trash: 'in', the: 'wind' };
    beforeEach(() => {
      mapped = mapStateToProps(testState);
    });
    test('showBulkManagement from root.showBulkManagement', () => {
      expect(mapped.showBulkManagement).toEqual(selectors.root.showBulkManagement(testState));
    });
  });
  describe('mapDispatchToProps', () => {
    test('initializeApp from thunkActions.app.initialize', () => {
      expect(mapDispatchToProps.initializeApp).toEqual(thunkActions.app.initialize);
    });
  });
});
