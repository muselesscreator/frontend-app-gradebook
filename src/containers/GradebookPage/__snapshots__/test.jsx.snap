// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GradebookPage component snapshot - shows BulkManagementTab if showBulkManagement 1`] = `
<WithSidebar
  sidebar={
    <GradebookFilters
      updateQueryParams={[MockFunction updateQueryParams]}
    />
  }
>
  <div
    className="px-3 gradebook-content"
  >
    <GradebookHeader />
    <Tabs
      defaultActiveKey="grades"
    >
      <Tab
        eventKey="grades"
        title="Grades"
      >
        <GradesTab
          updateQueryParams={[MockFunction updateQueryParams]}
        />
      </Tab>
      <Tab
        eventKey="bulk_management"
        title="Bulk Management"
      >
        <BulkManagementTab />
      </Tab>
    </Tabs>
  </div>
</WithSidebar>
`;

exports[`GradebookPage component snapshot - shows only GradesTab if showBulkManagement=false 1`] = `
<WithSidebar
  sidebar={
    <GradebookFilters
      updateQueryParams={[MockFunction updateQueryParams]}
    />
  }
>
  <div
    className="px-3 gradebook-content"
  >
    <GradebookHeader />
    <Tabs
      defaultActiveKey="grades"
    >
      <Tab
        eventKey="grades"
        title="Grades"
      >
        <GradesTab
          updateQueryParams={[MockFunction updateQueryParams]}
        />
      </Tab>
    </Tabs>
  </div>
</WithSidebar>
`;
